public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx

```diff
@@ -141,9 +140,10 @@ export function RichHistoryQueriesTab(props: Props) {
   const listOfDatasources = createDatasourcesList(datasourcesRetrievedFromQueryHistory);
 
   const listOfDatasourceFilters = datasourceFilters?.map(d => d.value);
-  const filteredQueriesByDatasource = datasourceFilters
-    ? queries?.filter(q => listOfDatasourceFilters?.includes(q.datasourceName))
-    : queries;
```

Comment:
Updated as new Select for empty selected list uses [] instead of null (which was used by LegacySelect)